src/shared/watchdog.c

```diff
@@ -17,8 +17,40 @@
 static int watchdog_fd = -1;
 static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
+static usec_t watchdog_pretimeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
+static int update_pretimeout(void) {
+        int r;
+
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY || watchdog_timeout == USEC_INFINITY)
```

Comment:
Should you check that watchdog_pretimeout is < watchdog_timeout? Otherwise the pretimeout interrrupt would never occur.

Reply:
You're right that would be good to check... The ioctl would return -EINVAL below in that case, but I'll add that check to make things clearer.